File: aemet/SergioDeIscar-aemet-java/SergioDeIscar-aemet-java/src/main/java/org/example/dto/InformeDto.java
package org.example.dto;

import org.example.models.Informe;
import org.example.models.PrecipitacionInforme;
import org.simpleframework.xml.Attribute;
import org.simpleframework.xml.Element;
import org.simpleframework.xml.Root;

import java.time.LocalDate;

@Root(name = "informe")
public class InformeDto {
    @Attribute(name = "day")
    private final String day;
    @Element(name = "tem_media")
    private final String temMedia;
    @Element(name = "tem_max")
    private final TemperaturaInformeDto temMax;
    @Element(name = "tem_min")
    private final TemperaturaInformeDto temMin;
    @Element(name = "precipitacion")
    private final PrecipitacionInformeDto precipitacion;

    public InformeDto(String day, String temMedia, TemperaturaInformeDto temMax, TemperaturaInformeDto temMin, PrecipitacionInformeDto precipitacion) {
        this.day = day;
        this.temMedia = temMedia;
        this.temMax = temMax;
        this.temMin = temMin;
        this.precipitacion = precipitacion;
    }

    // Getters
    public String getDay() {
        return day;
    }
    public String getTemMedia() {
        return temMedia;
    }
    public TemperaturaInformeDto getTemMax() {
        return temMax;
    }
    public TemperaturaInformeDto getTemMin() {
        return temMin;
    }
    public PrecipitacionInformeDto getPrecipitacion() {
        return precipitacion;
    }

    public Informe toClass(){
        return new Informe(LocalDate.parse(day), Double.parseDouble(temMedia), temMax.toClass(), temMin.toClass(), precipitacion.toClass());
    }
}

File: aemet/SergioDeIscar-aemet-kotlin/SergioDeIscar-aemet-kotlin/src/main/kotlin/dto/InformeDto.kt
package dto

import models.Informe
import org.simpleframework.xml.Attribute
import org.simpleframework.xml.Element
import org.simpleframework.xml.Root
import java.time.LocalDate
import java.time.LocalTime

@Root(name = "informe")
data class InformeDto(
    @field:Attribute(name = "day")
    @param:Attribute(name = "day")
    val day: String,

    @field:Element(name = "temMedia")
    @param:Element(name = "temMedia")
    val temMedia: String,

    @field:Element(name = "temMax")
    @param:Element(name = "temMax")
    val temMax: String,
    @field:Element(name = "poblacionMax")
    @param:Element(name = "poblacionMax")
    val poblacionMax: String,
    @field:Element(name = "timeMax")
    @param:Element(name = "timeMax")
    val timeMax: String,

    @field:Element(name = "temMin")
    @param:Element(name = "temMin")
    val temMin: String,
    @field:Element(name = "poblacionMin")
    @param:Element(name = "poblacionMin")
    val poblacionMin: String,
    @field:Element(name = "timeMin")
    @param:Element(name = "timeMin")
    val timeMin: String,

    @field:Element(name = "isPrecipitacion")
    @param:Element(name = "isPrecipitacion")
    val isPrecipitacion: String,
    @field:Element(name = "precipitacion")
    @param:Element(name = "precipitacion")
    val precipitacion: String
){
    fun toClass(): Informe{
        return Informe(
            day = LocalDate.parse(day),
            temMedia = temMedia.toDouble(),
            temMax = Triple(poblacionMax, LocalTime.parse(timeMax), temMax.toDouble()),
            temMin = Triple(poblacionMin, LocalTime.parse(timeMin), temMin.toDouble()),
            precipitacion = Pair(isPrecipitacion.toBoolean(), precipitacion.toDouble())
        )
    }
}
